### PR TITLE
8328560: java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java imports Applet

### PR DESCRIPTION
**Pull request by prrace**
This seems to have been some previous not very complete conversion of a test from applet to main.
It even still had Applet imported and miscellaneous comments. I've cleaned it up.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8328560](https://bugs.openjdk.org/browse/JDK-8328560): java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java imports Applet (**Bug** - P4)

### Reviewers
 * [Alexander Zvegintsev](https://openjdk.org/census#azvegint) (@azvegint - **Reviewer**)
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18386/head:pull/18386` \
`$ git checkout pull/18386`

Update a local copy of the PR: \
`$ git checkout pull/18386` \
`$ git pull https://git.openjdk.org/jdk.git pull/18386/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18386`

View PR using the GUI difftool: \
`$ git pr show -t 18386`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18386.diff">https://git.openjdk.org/jdk/pull/18386.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18386#issuecomment-2008344475)

### CODE REVIEW

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java [hunk 1]
```diff
@@ -22,115 +22,100 @@
  */
 
 /*
-  @test 1.2 98/08/05
```
Comment (azvegint) [inline test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java:25]: Now we don't have `@test` at all.

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java [hunk 2]
```diff
@@ -22,115 +22,100 @@
...
   @key headful
   @bug 4515763
   @summary Tests that clicking mouse and pressing keys generates correct amount of click-counts
-  @author andrei.dmitriev: area=awt.mouse
   @run main ClickDuringKeypress
 */
```
Comment (azvegint) [inline test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java:29]: We usually move this block just before a class declaration, for better readability.

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java [hunk 3]
```diff
@@ -22,115 +22,100 @@
...
+   static volatile Robot robot;
+   static volatile ClickDuringKeypress clicker;
+
+   public static void main(String[] args) throws Exception {
+       clicker = new ClickDuringKeypress();
+       try {
+           EventQueue.invokeAndWait(ClickDuringKeypress::createUI);
+           robot = new Robot();
+           robot.mouseMove(200, 200);
+           EventQueue.invokeAndWait(frame::show);
```
Comment (azvegint) [inline test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java:61]:
```suggestion
           EventQueue.invokeAndWait(() -> frame.setVisible(true));
```

`show` is deprecated.

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java [hunk 4]
```diff
@@ -22,115 +22,100 @@
...
+               if (!ready) {
+                   clicker.wait(5000);
+               }
+           } catch (InterruptedException ex) {
+           }
+           if (!ready) {
+               System.out.println("Not Activated. Test fails");
+               throw new RuntimeException("Not Activated. Test fails");
+           }
       }
```
Comment (azvegint) [inline test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java:96]: Do we really need this synchronization?
Shouldn't the standard 
```java
robot.waitForIdle();
robot.delay(1000);
```
be enough?

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java [hunk 5]
```diff
@@ -22,115 +22,100 @@
...
-      // the environment -- set the layout manager, add buttons,
-      // etc.
-
+   static volatile int lastClickCount = 0;
+   static volatile boolean clicked = false;
+   static volatile boolean ready = false;
+
+   static volatile Frame frame;
+   static volatile Robot robot;
+   static volatile ClickDuringKeypress clicker;
```
Comment (azvegint) [inline test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java:53]:
```suggestion
   static final ClickDuringKeypress clicker = new ClickDuringKeypress();
```
`clicker` can be made final

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java [hunk 6]
```diff
@@ -22,115 +22,100 @@
...
+      robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
       // Should trigger mouseClicked
       robot.keyRelease(KeyEvent.VK_B);
       robot.delay(1000);
 
       robot.setAutoDelay(DOUBLE_CLICK_AUTO_DELAY);
       for (int i = 0; i < CLICKCOUNT / 2; i++) {
-        robot.mousePress(InputEvent.BUTTON1_MASK);
+        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
         robot.delay(10);
```
Comment (azvegint) [inline test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java:111]: Here and after
```suggestion
```

We have already called `robot.setAutoDelay(DOUBLE_CLICK_AUTO_DELAY);` before. We can set `DOUBLE_CLICK_AUTO_DELAY` to 20 and remove the multiple `robot.delay(10)` calls.

Probably we also want to add `setAutoWaitForIdle(true)`

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java (hunk 1 above)
Comment (prrace) [reply test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java:25]: oops, fixed

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java (hunk 3 above)
Comment (prrace) [reply test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java]: yeah I know, I changed this as you suggested but
(1) I like the :: syntax, (2) I never thought it was necessary to deprecate show()

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java (hunk 4 above)
Comment (prrace) [reply test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java]: OK .. changed .. but I'm now a long way from removing an un-used import as a lot of this test is now quite different than before.

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java (hunk 6 above)
Comment (prrace) [reply test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java]: ok

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java (hunk 5 above)
Comment (prrace) [reply test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java]: ok

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java (hunk 2 above)
Comment (prrace) [reply test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java]: ok

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java [hunk 7]
```diff
@@ -21,116 +21,91 @@
...
- */
-
-import java.applet.Applet;
-import java.awt.*;
-import java.awt.event.*;
-
-public class ClickDuringKeypress implements MouseListener
- {
+public class ClickDuringKeypress implements MouseListener {
    //Declare things used in the test, like buttons and labels here
```
Comment (kumarabhi006) [inline test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java:45]: Comments can be removed.

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java [hunk 8]
```diff
@@ -21,116 +21,91 @@
...
+           EventQueue.invokeAndWait(ClickDuringKeypress::createUI);
+           robot = new Robot();
+           robot.setAutoWaitForIdle(true);
+           robot.delay(2000);
+           robot.mouseMove(200, 200);
+           robot.delay(2000);
+           EventQueue.invokeAndWait(() -> frame.setVisible(true));
+           doTest();
+       } finally {
+           if (frame != null) {
```
Comment (kumarabhi006) [inline test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java:67]: null check should be done on EDT ?

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java [hunk 9]
```diff
@@ -21,116 +21,91 @@
...
-          }
-          if (!ready) {
-              System.out.println("Not Activated. Test fails");
-              throw new RuntimeException("Not Activated. Test fails");
-          }
+    static void doTest() throws Exception {
+       robot.waitForIdle();
+       robot.delay(1000);
+       if (!ready) {
+           System.out.println("Not Activated. Test fails");
```
Comment (kumarabhi006) [inline test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java:88]: Looks redundant.

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java [hunk 10]
```diff
@@ -21,116 +21,91 @@
...
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Robot;
+import java.awt.event.InputEvent;
+import java.awt.event.KeyEvent;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.awt.event.WindowEvent;
```
Comment (azvegint) [inline test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java:34]:
```suggestion
import java.awt.event.WindowEvent;
```

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java [hunk 11]
```diff
@@ -21,116 +21,91 @@
...
-      // etc.
-
+   final static int DOUBLE_CLICK_AUTO_DELAY = 20;
+   static volatile int lastClickCount = 0;
+   static volatile boolean clicked = false;
+   static volatile boolean ready = false;
+
+   static volatile Frame frame;
+   static volatile Robot robot;
+   static final ClickDuringKeypress clicker = new ClickDuringKeypress();
```
Comment (azvegint) [inline test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java:54]: On the other hand, we can get rid of the instantiation of this and MouseListener empty methods overrides class by simply adding this:

```java
frame.addMouseListener(new MouseAdapter() {
          @Override
          public void mouseClicked(MouseEvent e) {
              System.out.println(e.toString());
              clicked = true;
              lastClickCount = e.getClickCount();
          }
      });
```

Either way, I'm fine with it.

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java [hunk 12]
```diff
@@ -21,116 +21,91 @@
  * questions.
  */
 
+import java.awt.AWTException;
```
Comment (azvegint) [inline test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java:24]:
```suggestion
```
Unused

**Review by azvegint — APPROVED** [review]

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java (hunk 8 above)
Comment (prrace) [reply test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java:67]: Not needed for the null check

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java (hunk 7 above)
Comment (prrace) [reply test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java:45]: I thought about it already, seemed harmless, but I can remove it.

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java (hunk 10 above)
Comment (prrace) [reply test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java:34]: fixed

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java (hunk 12 above)
Comment (prrace) [reply test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java:24]: right not needed now

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java (hunk 9 above)
Comment (prrace) [reply test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java:88]: good to log, so I think it should stay

File: test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java (hunk 11 above)
Comment (prrace) [reply test/jdk/java/awt/event/MouseEvent/ClickDuringKeypress/ClickDuringKeypress.java:54]: I thought about doing something like this .. but decided to stop the creep of changes.
So I'll leave it.

**Review by kumarabhi006 — APPROVED** [review]

**Review by azvegint — APPROVED** [review]

**Pull request closed without merging**